File: signature/nginx.go
package signature

var NginxSignature = Signature{
	Name:        "Nginx",
	Description: "An HTTP web server, reverse proxy, content cache, load balancer, TCP/UDP proxy server, and mail proxy server.",
	Cpe:         "cpe:/a:nginx:nginx",

	DetectPattern: DetectPattern{
		Headers: []Header{{
			Name:  "server",
			Value: "nginx",
		}},
	},
	VersionPattern: VersionPattern{
		Headers: []Header{{
			Name:  "server",
			Value: `nginx/(\d+\.\d+\.\d+)`,
		}},
	},

	OnlySameHost: true,
}

File: src/signatures/nginx.ts
import type { Signature } from "./_types.js";

export const nginxSignature: Signature = {
  name: "Nginx",
  description:
    "An HTTP web server, reverse proxy, content cache, load balancer, TCP/UDP proxy server, and mail proxy server.",
  cpe: "cpe:/a:nginx:nginx",
  rules: [
    {
      confidence: "high",
      headers: {
        server: { regex: "^nginx/?(\\d+\\.\\d+\\.\\d+)?" },
      },
    },
  ],
};
